sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -308,6 +308,12 @@ private class ScriptTransformationWriterThread(
       }
       threwException = false
     } catch {
+      // TaskKilledException should not be thrown again, otherwise it will be captured by
+      // SparkUncaughtExceptionHandler, then Executor will exit because of TaskKilledException.
+      case e: TaskKilledException =>
```

Comment:
which line of code may throw this exception? and how is this exception handled in other operators?

Reply:
Thanks for your reply @cloud-fan , the error log and problem analysis of one bad case I paste in https://issues.apache.org/jira/browse/SPARK-25158, in that case `.UnsafeInMemorySorter$SortedIterator.loadNext` throw `TaskKilledException`, and in another case `TaskKilledException` throw by `InterruptibleIterator.hasNext` , and from the error stack line 281 `iter.map(outputProjection).foreach ` is the stack root.

Reply:
Actually ` ScriptTransformationExec`  like a streaming pipe, but other operators basically no such characteristics.

Reply:
On the other hand perhaps `ScriptTransformationWriterThread` should not throw any Exception because the Exception mark and handle by `ScriptTransformationExec` , but I'm not sure.

Reply:
> Actually ScriptTransformationExec like a streaming pipe, but other operators basically no such characteristics.

can you elaborate on it? How `TaskKilledException` is handled in other operators?

Reply:
I see. So `ScriptTransformationExec` is special because
1. it starts a new thread
2. the new thread is very likely to throw `TaskKilledException`, when speculation is on.

I think we should not kill the executor just because `ScriptTransformationWriterThread` fails. We should log the error, instead of throwing it.

Reply:
Cannot argree more ~